src/runtime/pkg/kata-monitor/metrics.go

```diff
@@ -230,9 +237,19 @@ func (km *KataMonitor) aggregateSandboxMetrics(encoder expfmt.Encoder) error {
 	}
 
 	// write metrics to response.
-	for _, mf := range metricsMap {
-		if err := encoder.Encode(mf); err != nil {
-			return err
+	for fullName, mf := range metricsMap {
```

Comment:
First iterate over `filterFamilies`(a slice) and then `metricsMap`(a map) maybe efficient.